### PR TITLE
fix: remove PR if max attempt reached

### PR DESCRIPTION
**Pull request by m-rudyk**
# Pull Request

## Description

Add increment to Queue processing, when PR processing interrupted, it should be kept in queue until max scanning attempts reached. 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code cleanup/refactoring
- [ ] Documentation update
- [ ] This change requires a documentation update
- [ ] CI system update
- [x] Test Coverage update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**Test Configuration**:
* Java: v17
* LPVS Release: v1.x.x

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] My code meets the required code coverage for lines (90% and above)
- [x] My code meets the required code coverage for branches (80% and above)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/main/java/com/lpvs/service/LPVSQueueService.java [hunk 1]
```diff
@@ -271,7 +271,23 @@ public void processWebHook(LPVSQueue webhookConfig) {
             pullRequest = lpvsPullRequestRepository.saveAndFlush(pullRequest);
             log.error("Can't authorize commentResults() " + e);
             e.printStackTrace();
-            delete(webhookConfig);
+            int currentAttempts = webhookConfig.getAttempts();
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSQueueService.java:274]: Please check lines 149-152.
```
if (webhookConfigList.size() > 0) {
    LPVSQueue webhookConfig = getLatestScan(webhookConfigList);
    webhookConfig.setAttempts(webhookConfig.getAttempts() + 1);
    queueRepository.save(webhookConfig);
}
```

Do we need this code?

File: src/main/java/com/lpvs/service/LPVSQueueService.java [hunk 2]
```diff
@@ -146,14 +146,9 @@ public BlockingDeque<LPVSQueue> getQueue() {
...
         List<LPVSQueue> webhookConfigList = queueRepository.getQueueList();
-        if (webhookConfigList.size() > 0) {
-            LPVSQueue webhookConfig = getLatestScan(webhookConfigList);
-            webhookConfig.setAttempts(webhookConfig.getAttempts() + 1);
-            queueRepository.save(webhookConfig);
-        }
         for (LPVSQueue webhook : webhookConfigList) {
             log.info("PROCESSING WebHook id = " + webhook.getId());
-            if (webhook.getAttempts() > maxAttempts) {
+            if (webhook.getAttempts() < maxAttempts) {
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSQueueService.java:151]: Why "<"?
This is the processing of the case when webhook.getAttempts() > maxAttempts.

File: src/main/java/com/lpvs/service/LPVSQueueService.java [hunk 3]
```diff
@@ -146,14 +146,9 @@ public BlockingDeque<LPVSQueue> getQueue() {
...
                 LPVSPullRequest pullRequest = new LPVSPullRequest();
                 pullRequest.setPullRequestUrl(webhook.getPullRequestUrl());
                 pullRequest.setUser(webhook.getUserId());
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSQueueService.java:154]: The same webhook processing should be added to exception processing before delete().

File: src/main/java/com/lpvs/service/LPVSQueueService.java [hunk 4]
```diff
@@ -258,20 +225,36 @@ public void processWebHook(LPVSQueue webhookConfig) {
                 log.debug("Creating comment");
                 gitHubService.commentResults(webhookConfig, files, detectedConflicts, pullRequest);
                 log.debug("Results posted on GitHub");
+                delete(webhookConfig);
             } else {
                 log.warn("Files are not found. Probably pull request is not exists.");
                 gitHubService.commentResults(webhookConfig, null, null, pullRequest);
                 delete(webhookConfig);
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSQueueService.java:232]: This delete is incorrect. Please remove it.

File: src/main/java/com/lpvs/service/LPVSQueueService.java [hunk 5]
```diff
@@ -258,20 +230,35 @@ public void processWebHook(LPVSQueue webhookConfig) {
...
-                delete(webhookConfig);
                 throw new Exception(
                         "Files are not found. Probably pull request does not exist. Terminating.");
             }
-            delete(webhookConfig);
         } catch (Exception | Error e) {
             pullRequest.setStatus(LPVSPullRequestStatus.INTERNAL_ERROR.toString());
             pullRequest = lpvsPullRequestRepository.saveAndFlush(pullRequest);
             log.error("Can't authorize commentResults() " + e);
             e.printStackTrace();
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSQueueService.java:244]: Please change to: 
```suggestion
            log.error("Can't authorize commentResults() " + e.getMessage());
```

File: src/main/java/com/lpvs/service/LPVSQueueService.java [hunk 6]
```diff
@@ -258,20 +230,35 @@ public void processWebHook(LPVSQueue webhookConfig) {
...
+                    log.warn("Failed to update Queue element");
+                }
+                queueRepository.save(webhookConfig);
+            } else {
+                log.warn(
+                        "Maximum amount of processing webhook reached for pull request: "
+                                + pullRequest.getId()
+                                + " "
+                                + pullRequest.getPullRequestUrl());
+                delete(webhookConfig);
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSQueueService.java:260]: Before deleting webhook, we have to post results on GitHub:
```suggestion
                delete(webhookConfig);
                gitHubService.commentResults(webhookConfig, null, null, pullRequest);
```

File: src/main/java/com/lpvs/service/LPVSQueueService.java [hunk 7]
```diff
@@ -258,20 +230,35 @@ public void processWebHook(LPVSQueue webhookConfig) {
                 log.debug("Creating comment");
                 gitHubService.commentResults(webhookConfig, files, detectedConflicts, pullRequest);
                 log.debug("Results posted on GitHub");
+                delete(webhookConfig);
             } else {
                 log.warn("Files are not found. Probably pull request is not exists.");
                 gitHubService.commentResults(webhookConfig, null, null, pullRequest);
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSQueueService.java:236]: We don't need to post results whenever we have exceptions.
```suggestion
```

**Review by t-naumenko — APPROVED** [review]
LGTM

**Comment by o-kopysov** [comment]
I've verified the code on local PC:
image

**Review by o-kopysov — APPROVED** [review]
Verified.

**Pull request merged**